### PR TITLE
Drop defaults from environment variables

### PR DESCRIPTION
**Pull request by slewis74**
Upon reflection, having the client pick up connection details from environment variables wasn't a great design choice. We've had a number of issues and odd behaviours out of variables being picked up unexpectedly.

This PR drops the defaults, it will be the caller's responsibility to apply defaults that make sense for them.

Also as part of this PR the create release/executions API operations have been updated so that callers only need to know the spaceName and not the spaceId. The spaceId is resolved internally and added to the objects being passed to the server.

A number of the repositories (the ones the GHAs use to retrieve additional data) have also been converted to use the new model

### CODE REVIEW

File: src/clientConfiguration.test.ts [hunk 1]
```diff
@@ -1,28 +1,52 @@
-import { ClientConfiguration, processConfiguration } from "./clientConfiguration";
+import { ClientConfiguration } from "./clientConfiguration";
+
+export const EnvironmentVariables = {
+    ApiKey: "OCTOPUS_TEST_API_KEY",
+    URL: "OCTOPUS_TEST_URI",
```
Comment (domenicsim1) [inline src/clientConfiguration.test.ts:5]: should this be updated to `URI` or `InstanceURI`?

**Review by domenicsim1 — APPROVED** [review]
Looks good, left a question.

File: src/clientConfiguration.test.ts (hunk 1 above)
Comment (domenicsim1) [reply src/clientConfiguration.test.ts:5]: actually, should we also make them camelcase too?

**Comment by slewis74** [comment]
Replaced by #126

**Pull request closed without merging**